client/ping.go

```diff
@@ -13,18 +13,15 @@ func (cli *Client) Ping(ctx context.Context) (types.Ping, error) {
 		return ping, err
 	}
 	serverResp, err := cli.doRequest(ctx, req)
-	if err != nil {
-		return ping, err
-	}
-	defer ensureReaderClosed(serverResp)
+	if serverResp.header != nil {
```

Comment:
So, I agree a fair amount of the handling in `doRequest` doesn't even make sense in the client code and should be handled by the CLI instead.

However I think this nil check would still be required. I'm also not sure it's right to refactor the whole client lib in order to fix this issue with ping details.